source/common/router/BUILD

```diff
@@ -333,6 +333,7 @@ envoy_cc_library(
         "//source/common/protobuf:utility_lib",
         "@envoy_api//envoy/config/core/v3:pkg_cc_proto",
         "@envoy_api//envoy/config/route/v3:pkg_cc_proto",
+        "@envoy_api//envoy/extensions/filters/http/ratelimit/v3:pkg_cc_proto",
```

Comment:
That's not good. core components shouldn't depend on extensions..

Reply:
Ok, I see. so yeah, that's no bueno, core code can't depend on extensions. To resolve this issue in an even better way, create source/extensions/filters/http/ratelimit/ratelimit_policy_impl.h and .cc

There you can declare new classes that extend all the classes you are making new constructors for, and declare/implement new constructors there. You'd need to make private properties protected temporary.

After the whole router ratelimit logic is removed, all the code implementing RateLimitPolicy will go there.

Reply:
Gotta remove this now right?